### ISSUE TITLE
Add docs section or page describing how to tell toolbox to build customized images

### ISSUE THREAD

**Issue by djasa** [containers/toolbox#1102]
**Is your feature request related to a problem? Please describe.**
Toolbx creates images that are expected to be modified. However many people are likely to need stable set of their modification that'd be repeated on every (re)build of their toolboxes.

**Describe the solution you'd like**
There are nice solutions available online:
  * [lisbox](https://github.com/allisonkarlitskaya/lisbox)
  * [build-arch-toolbox](https://gitlab.com/garrett/silverblue/-/blob/master/build-arch-toolbox)
  * [build-dev-toolbox](https://gitlab.com/garrett/silverblue/-/blob/master/build-devtoolbox)

I'd like to see example like this explained in online docs and manual page in a way that person who never worked with containers before will be able to write their own basic script that'd install custom set of packages and modify some stuff in toolbox container's /etc.

**Describe alternatives you've considered**
-

**Additional context**
This could be also reply to requests for other distro's support for the time being.

**Comment by akdev1l** [comment]
Hi I will leave some observations here,

I am not sure whether the documentation really belongs in `toolbox` itself. There's no `toolbox build` command and I think that's good. `toolbox` is by design supposed to be less complex than `podman`.

The reality of building a custom image is that you *need* to know `Containerfile` syntax - this is going to be the biggest hurdle to any new user. Documenting such syntax just seems out of scope as there are plenty of resources online doing that. 

I guess we could create some sort of tutorial that shows a minimum of what needs to be done(I still don't think this should be in the man pages however). Though I have been thinking of creating a small web application that essentially allows you to quickly generate a Containerfile without knowing the syntax - maybe that would be a better approach. Let me know what you think about this approach.

As for: 

> This could be also reply to requests for other distro's support for the time being.

I went ahead myself and build images for some popular distros here: https://github.com/akdev1l/toolbox-images. Feel free to give it a try.

This issue seems closely related (if not duplicate) with  https://github.com/containers/toolbox/issues/1077 - I left a quick way of creating a custom image without resorting to Containerfiles there. Maybe that will work for you for the time being.

just my 2 cents, hope that's somewhat useful

**Comment by djasa** [comment]
> Hi I will leave some observations here,

Thanks :)

> I am not sure whether the documentation really belongs in `toolbox` itself. There's no `toolbox build` command and I think that's good. `toolbox` is by design supposed to be less complex than `podman`.
> 
> The reality of building a custom image is that you _need_ to know `Containerfile` syntax - this is going to be the biggest hurdle to any new user. Documenting such syntax just seems out of scope as there are plenty of resources online doing that.

I don't think that full reference of Dockerfile (or `buildah*` commands?) would be necesary here. Just bare minimum allowing users to do the basic tasks (image base, run command, copy file) with pointers to podman or Dockerfile resources where this isn't enough.

> This issue seems closely related (if not duplicate) with https://github.com/containers/toolbox/issues/1077 - I left a quick way of creating a custom image without resorting to Containerfiles there.

Thanks, my search didn't yield it. It seems to me that the ends and means are different in these two however. Creating a base image is good when you need more images in parallel, however for how I use toolbox (sth breaks within containers after months of having it), such an approach wouldn't be wise as after just a few iterations, I'd end up with image based on old system and full of CVEs. :/

**Comment by travier** [comment]
See also the progress we made in https://github.com/containers/toolbox/issues/1019 for other distro toolboxes.

**Comment by matthiasclasen** [comment]
After looking around a bit in the existing documentation, I would suggest that we add some information to the website with details about toolbox images, and how to create custom ones.

This does not seem to fit well into either the man pages or the existing install & use section.

**Comment by matthiasclasen** [comment]
https://github.com/containers/toolbox/issues/1077 explains a nice way to create custom images that would be good to include in the docs

**Comment by djasa** [comment]
@matthiasclasen hi, this is what I meant with just one tiny bit missing: command how to create a toolbx image from Containerfile (I guess `podman build -f Containerfile`? Will the resulting image be directly visible/usable by toolbox?).

**Comment by akdev1l** [comment]
> @matthiasclasen hi, this is what I meant with just one tiny bit missing: command how to create a toolbx image from Containerfile (I guess `podman build -f Containerfile`? Will the resulting image be directly visible/usable by toolbox?).

The command you want is `podman build -t mycustomimage:latest .` and then it will be usable by toolbox by doing `toolbox create -i mycustomimage:latest`.

**Comment by djasa** [comment]
> The command you want is podman build -t mycustomimage:latest . and then it will be usable by toolbox by doing toolbox create -i mycustomimage:latest.

Thanks!

@matthiasclasen, could we have these spelled out in [/install/](https://containertoolbx.org/install/) or alongside [/example-container-file/](https://containertoolbx.org/example-container-file/), please?
